Update the list of build types

daniel-larraz opened the pull request.

aniemetz reviewed (changes requested):
I don't think this is right. `safe-mode` is an option, not a build type, right?

daniel-larraz commented:
> I don't think this is right. `safe-mode` is an option, not a build type, right?

Actually, `safe-mode` is a build type, not an option. It behaves exactly like the `production` build type, with the key difference being that the `--safe-options` flag is enabled (`true`) by default (see [here](https://github.com/cvc5/cvc5/blob/main/configure.sh#L340)).

aniemetz reviewed (commented):
It seems that currently, we don't have any documentation of the build types in the help message of the configure script. I think especially now that we have added safe-mode, it would be good to add documentation.

ajreynol commented:
I agree, notice we aren't advertizing `safe-mode` in the release notes anymore, so perhaps this could wait?

daniel-larraz commented:
I added a brief description of each build type to the help message in the configure script. I also removed the `competition-inc` build type, as it was no longer in use.
